### PR TITLE
Add support for parsing sensor status

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: device/src/drivers/ble/mesh/model/sensor/mod.rs [hunk 1]
```diff
@@ -82,7 +82,9 @@ pub trait SensorConfig: Clone {
 }
 
 pub trait SensorData {
-    fn decode(&mut self, property: PropertyId, data: &[u8]) -> Result<(), ParseError>;
+    fn decode(property: PropertyId, data: &[u8]) -> Result<Self, ParseError>
```
Comment (lulf) [inline device/src/drivers/ble/mesh/model/sensor/mod.rs:85]: Looking at how this is being used now, SensorData can't have more than 1 property, which won't work in the general case. I think we need to modify the trait back and add a default constraint:
```
pub trait SensorData: Default {
    fn decode(&mut self, property: PropertyId, data: &[u8]) -> Result<(), ParseError>
}
```

File: device/src/drivers/ble/mesh/model/sensor/mod.rs [hunk 2]
```diff
@@ -625,6 +630,26 @@ where
...
+
+    fn parse(parameters: &[u8]) -> Result<Self, ParseError> {
+        let mut pos = 0;
+        let mut result = None;
+        for d in C::DESCRIPTORS {
+            pos += 2;
+            let parameters = &parameters[pos..(pos + d.size)];
+
+            result = Some(C::Data::decode(d.id, parameters));
+        }
```
Comment (lulf) [inline device/src/drivers/ble/mesh/model/sensor/mod.rs:642]: Problem here is that parsing sensor data with more than 1 property becomes impossible. But with the proposed trait above, this code becomes:

```
let mut data = Default::default();
for d in C::DESCRIPTORS {
    // Decode property id and verify match
    let parameters = ...
    data.decode(id, parameters)?;
}

File: device/src/drivers/ble/mesh/model/sensor/mod.rs [hunk 3]
```diff
@@ -625,6 +630,26 @@ where
     pub fn new(data: C::Data<'a>) -> Self {
         Self { data }
     }
+
+    fn parse(parameters: &[u8]) -> Result<Self, ParseError> {
+        let mut pos = 0;
+        let mut result = None;
+        for d in C::DESCRIPTORS {
```
Comment (lulf) [inline device/src/drivers/ble/mesh/model/sensor/mod.rs:637]: If you look at the 'encode', the property descriptor can be either 2 or 3 bytes. The decoded value should also be verified that it's matching `d`.

**Review by lulf — COMMENTED** [review]
Great work! Left some comments.

File: device/src/drivers/ble/mesh/model/sensor/mod.rs (hunk 1 above)
Comment (dejanb) [reply device/src/drivers/ble/mesh/model/sensor/mod.rs]: Yeah, I wasn't happy with that solution. I reworked it to use Default trait and now it looks much nicer.

File: device/src/drivers/ble/mesh/model/sensor/mod.rs (hunk 3 above)
Comment (dejanb) [reply device/src/drivers/ble/mesh/model/sensor/mod.rs:636]: I think we're missing a bit logic in properly parsing PropertyID ... Ideally, I'd just do (pseudo code)

```
if PropertyId::parse(params) != d.id {
    return Err(ParseError::InvalidValue);
}
```

but currently `parse()` assumes two bytes and doesn't work for my use case.

I'll take this after the break in this or separate PR.

Comment (dejanb) [reply device/src/drivers/ble/mesh/model/sensor/mod.rs:635]: @lulf this is now implemented. Property ids are properly parsed and checked.

**Pull request merged**